File: Python Word Examples/ProtectionApiDemo/protectDocument.py
import spirecloudword
from spirecloudword.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudword.api.ProtectionApi(configuration)

file_name = 'Template.docx'
#supports types: AllowOnlyReading/AllowOnlyComments/AllowOnlyFormFields/AllowOnlyRevisions/NoProtection
protection_type = 'AllowOnlyReading'
extra = {'folder': "input", 'protectiong_pwd': '123456', 'dest_file_path': 'output/ProtectDocument_AllowOnlyReading.docx'}
result =api.protect_document(file_name, protection_type, **extra)

File: JavaScript Word Examples/ProtectionApiDemo/protectDocument.js
(function (){
    var appId = "your id";
    var appKey = "your key";
    var baseUrl = "https://api.e-iceblue.cn";

    var Spirecloudword = require('../../src/index');
    var apiClient = new Spirecloudword.ApiClient(appId, appKey,baseUrl);
    var instance = new Spirecloudword.ProtectionApi(apiClient);

    var filename = "ProtectDocument.docx";
    //available types: AllowOnlyReading, AllowOnlyComments, AllowOnlyFormFields, AllowOnlyRevisions, NoProtection
    var protectionType = "AllowOnlyReading";
    var opts = {
        folder: 'input',
        //docPassword: '123', //if input document has original password
        protectiongPwd: '123456'
    };
    var destFilePath = 'output/ProtectDocument_output.docx';
    instance.protectDocument(filename, protectionType, destFilePath, opts, function (error, data, response) {
        if (error) throw error;  
    });
})();